#1)Escribe un programa que intente dividir dos números. Si el segundo número es cero,c
# aptura la excepción ZeroDivisionError y muestra un mensaje de error al usuario.



try:
    Cuenta = 20/0
    print(Cuenta)
except ZeroDivisionError :
  print("No se puede dividir por cero")

#2)Escribe un programa que intente sumar un número y una cadena.
#  Si se produce un errorde tipo, captura la excepción TypeError y muestra un mensaje de error al usuario


try:
    Cuenta = 20 + 'A'
    print(Cuenta)
except TypeError :
    print("No se puede combinar letras con números")


#3)Escribe un programa que intente acceder a una clave que no existe en undiccionario.
#  Si se produce una excepción KeyError, captura la excepción y muestra



try:
    Usuario = {"Nombre" : "Jose", "Edad": "25", "Emails" :"josepepe@gmail.com" }
    print (Usuario["Telefono"])
except KeyError :
       print("No se pudo encontrar el telefono ")


#4)Escribe un programa que intente abrir un archivo que no existe. Si se produce una excepciónFileNotFoundError,
#captura la excepción y muestra un mensaje de error al usuario. Sinembargo, también intenta crear el archivo si no existe.     
 
try:
    Archivo = open("archivo_inexistente.txt", "r")
    print(Archivo)
except FileNotFoundError:
    print("Error: El archivo no existe.")

# Intentar crear el archivo
try:
    Archivo =open ("archivo_inexistente.txt", "w")
    print(Archivo)
except Exception as e :
     print(f"Se produjo un error al intentar crear el archivo: {e}")
#finally:
#    print("Error: El archivo se creo correctamente.")
#    Archivo.close()
#    print("El archivo se cerro exitosamente")

try:
    Archivo = input("Ingrese el nombre del archivo: ")
    print(Archivo)
except FileNotFoundError:
    print("Error: El archivo no existe.")



#5)Escribe un programa que intente dividir dos números. Si el segundo número es cero,captura la excepción ZeroDivisionError.
#Si el primer número es un número no válido,captura la excepción ValueError. En cualquier caso, muestra un mensaje de error al usuario.

try:
   numero = int(input("Ingrese el numero: "))
   numero2 = int(input("Ingrese el numero: "))
   cuenta = numero/numero2
except ZeroDivisionError :
    print("no se puede dividir por cero")
except ValueError as ve:
 print("El valor ingresado no es un numero")   
